addons/im_livechat/models/res_partner.py

```diff
@@ -79,3 +89,14 @@ def _compute_display_name(self):
             return
         for partner in self:
             partner.display_name = partner.name
+
+    def action_view_livechat_sessions(self):
+        action = self.env.ref('im_livechat.discuss_channel_action').read()[0]
+        channel_ids = self.env['im_livechat.channel.member.history'].search([
+            ('partner_id', 'in', self.ids)
+        ]).mapped('channel_id').ids
```

Comment:
- `mapped` is not required here.
- We might add an inverse field on discuss_channel to avoid those two queries before executing the action.